server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByWithValueFilterDataSet.java

```diff
@@ -122,13 +127,18 @@ protected RowRecord nextWithoutConstraint() throws IOException {
 
     long[] timestampArray = new long[timeStampFetchSize];
     int timeArrayLength = 0;
-    if (hasCachedTimestamp) {
+    while (!cachedTimestamps.isEmpty()) {
+      long timestamp = cachedTimestamps.remove();
       if (timestamp < curEndTime) {
+        if (!groupByTimePlan.isAscending() && timestamp < curStartTime) {
+          cachedTimestamps.addFirst(timestamp);
+          return constructRowRecord(aggregateResultList);
+        }
         if (timestamp >= curStartTime) {
-          hasCachedTimestamp = false;
           timestampArray[timeArrayLength++] = timestamp;
         }
       } else {
+        cachedTimestamps.addFirst(timestamp);
```

Comment:
It seems that this timestamp is useless, we don't need to put it back to `cachedTimestamps`.

Reply:
i think we need put it back , for example , the interval is [0,10) , [10,20), [20,30), if the first cache is 30 ,and the end time is 10, so that need put back